qa/vagrant/src/main/java/org/elasticsearch/packaging/test/WindowsServiceTestCase.java

```diff
@@ -0,0 +1,209 @@
+import static org.elasticsearch.packaging.util.Archives.verifyArchiveInstallation;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public abstract class WindowsServiceTestCase extends PackagingTestCase {
+
+    private static final String DEFAULT_ID = "elasticsearch-service-x64";
+    private static final String DEFAULT_DISPLAY_NAME = "Elasticsearch " + FileUtils.getCurrentVersion() + " (elasticsearch-service-x64)";
+    private static String serviceScript;
+
+    private Shell sh;
+
+    @Before
+    public void createShell() {
+        sh = new Shell();
+    }
+
+    @BeforeClass
+    public static void ensureWindows() {
+        assumeTrue(Platforms.WINDOWS);
+    }
+
+    @After
+    public void uninstallService() {
+        sh.runIgnoreExitCode(serviceScript + " remove");
+    }
+
+    private Result runWithoutJava(String script) {
+        // on windows, removing java from PATH and removing JAVA_HOME is less involved than changing the permissions of the java
+        // executable. we also don't check permissions in the windows scripts anyway
+        final String originalPath = sh.run("$Env:PATH").stdout.trim();
+        final String newPath = Arrays.stream(originalPath.split(";"))
+            .filter(path -> path.contains("Java") == false)
+            .collect(joining(";"));
+
+        // note the lack of a $ when clearing the JAVA_HOME env variable - with a $ it deletes the java home directory
+        // https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/providers/environment-provider?view=powershell-6
+        //
+        // this won't persist to another session so we don't have to reset anything
+        return sh.runIgnoreExitCode(
+            "$Env:PATH = '" + newPath + "'; " +
+                "Remove-Item Env:JAVA_HOME; " +
+                script
+        );
+    }
+
+    public void test10InstallArchive() {
+        installation = installArchive(distribution());
+        verifyArchiveInstallation(installation, distribution());
+        serviceScript = installation.bin("elasticsearch-service.bat").toString();
+    }
+
+    public void test11InstallServiceExeMissing() throws IOException {
+        Path serviceExe = installation.bin("elasticsearch-service-x64.exe");
+        Path tmpServiceExe = serviceExe.getParent().resolve(serviceExe.getFileName() + ".tmp");
+        Files.move(serviceExe, tmpServiceExe);
+        Result result =  sh.runIgnoreExitCode(serviceScript + " install");
+        assertThat(result.exitCode, equalTo(1));
+        assertThat(result.stdout, containsString("elasticsearch-service-x64.exe was not found..."));
+        Files.move(tmpServiceExe, serviceExe);
+    }
+
+    public void test12InstallService() {
+        sh.run(serviceScript + " install");
+        assertService(DEFAULT_ID, "Stopped", DEFAULT_DISPLAY_NAME);
+        sh.run(serviceScript + " remove");
```

Comment:
I think these should get taken care of by `uninstallService`? Or is the point here that we want to assert exit code 0 when uninstalling it in these tests

Reply:
Yeah the point is we want exit code 0 here.